### PR TITLE
Fixes #29849 - Add DSL docs for hosts

### PR DESCRIPTION
**Pull request by ofedoren**
Unfortunately, apipie-dsl doesn't have a way to merge concerns' docs into the main class. So, for now I suggest have documentation in separate classes with some kind of note for usage.

### CODE REVIEW

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -97,6 +102,64 @@ def build_hooks
 
   graphql_type '::Types::Host'
 
+  apipie :class, 'A class representing managed Host object' do
+    prop_group :basic_model_props, ApplicationRecord, meta: { friendly_name: 'host', name_desc: 'Host FQDN, e.g. my-host.example.com' }
+    property :architecture, String, desc: 'Returns a string representing the architecture of the host (e.g. "x86_64") or an empty string if no architecture is assigned (unmanaged host)'
```
Comment (shiramax) [inline app/models/host/managed.rb]: It returns Architecture obj and not a string.

File: app/models/host/managed.rb [hunk 2]
```diff
@@ -97,6 +102,64 @@ def build_hooks
...
+    property :build?, one_of: [true, false], desc: 'Returns true if the host is pending for build, false otherwise'
+    property :certname, String, desc: 'Returns a name used in puppet certificate, this is usually either equal to FQDN or random UUID if `use_uuid_for_certificates` setting is enabled'
+    property :compute_resource, 'ComputeResource', desc: 'Returns a compute resource object the host exists in, nil if no compute resource is assigned (e.g. baremetal host)'
+    property :domain, 'Domain', desc: 'Returns a domain object the host primary interface belongs to, nil if no domain is assigned (unmanaged host)'
+    property :environment, 'Environment', desc: 'Returns a string representing the puppet environment the host is assigned to (e.g. "production") or an empty string if no puppet environment is assigned'
+    property :hostgroup, 'Hostgroup', desc: 'Returns a host group object the host is assigned to, nil if no host group is assigned'
+    property :interfaces, Array, desc: 'Returns an array of all host interfaces objects'
```
Comment (shiramax) [inline app/models/host/managed.rb]: array_of: ['Nic::Managed']

File: app/models/host/managed.rb [hunk 3]
```diff
@@ -97,6 +102,64 @@ def build_hooks
...
+    property :root_pass, String, desc: "Returns host's encrypted password hash"
+    property :use_image, one_of: [true, false], desc: 'Returns whether provisioning is image based'
+    property :sp_name, String, desc: "Returns BMC's NIC name"
+    property :sp_ip, String, desc: 'Returns BMC\'s NIC IP'
+    property :sp_mac, String, desc: 'Returns BMC\'s NIC MAC'
+    property :sp_subnet, String, desc: "Returns BMC's NIC subnet"
+    property :jumpstart_path, String, desc: 'Calculates the jumpstart\'s path in relation to the domain and convert host to an IP'
+    property :install_path, String, desc: 'Calculates the media\'s path in relation to the domain and convert host to an IP'
+    property :image_build?, one_of: [true, false], desc: 'Returns true if this host provision method is image, meaning image based provisioning, false otherwise'
+    property :medium, String, desc: 'Returns installation medium\'s name associated with the host'
```
Comment (shiramax) [inline app/models/host/managed.rb]: it returns Medium object.

**Review by shiramax — CHANGES_REQUESTED** [review]

**Comment by ofedoren** [comment]
Thanks, @shiramax! Updated.

**Comment by shiramax** [comment]
[test foreman]

**Comment by shiramax** [comment]
@ofedoren I think the tests are related.

**Comment by ofedoren** [comment]
@shiramax, thank you! I didn't notice one change that somehow migrated into this PR, should be fixed now.

UPD: all seems to be :green_heart:

**Review by shiramax — APPROVED** [review]

**Comment by shiramax** [comment]
thanks @ofedoren !

**Pull request merged**